### DISCUSSION TITLE
Merge queue recheck

### DISCUSSION THREAD

**Discussion by devportilla** [community/community#16089] [Pull Requests]
Adding a pull request with all checks passed to the queue, triggers the checks again once in the queue, delaying the merging to the time of those checks. Since the button only is "Merge when ready" there is no chance to merge right away the pull request. Suggestion is to merge right away the pull request if all checks passed and branch is updated.

**Comment by christian-zyweck** [comment]
This is currently a really annoying issue on my clients main repository, because the checks take up a lot of time due to a large number of tests. From our perspective, it should not be necessary to re-run all checks on a pull request that is the only item in the queue, especially if it has been rebased prior to adding it to the queue.
